pkg/api/types.go

```diff
@@ -1829,3 +1811,7 @@ func FormatProdFQDNByLocation(fqdnPrefix string, location string, cloudName stri
 	FQDNFormat := AzureCloudSpecEnvMap[targetEnv].EndpointConfig.ResourceManagerVMDNSSuffix
 	return fmt.Sprintf("%s.%s."+FQDNFormat, fqdnPrefix, location)
 }
+
+func (p *Properties) isAzureStack() bool {
+	return p.OrchestratorProfile != nil && p.OrchestratorProfile.KubernetesConfig != nil && p.IsAzureStackCloud()
```

Comment:
I'm actually a little confused by this method now. What is the purpose of checking:

- non-nil OrchestratorProfile
- non-nil KubernetesConfig
- Azure Stack cloud definition

all together?

Why not just invoke the existing `p.IsAzureStackCloud()` method in the addons enabled methods?

Reply:
I see what's going on. Sigh. These "is addons enabled" methods should not be a method of the `Properties` object. Stand by.

Reply:
they weren't. That was to accommodate using the `isAzureStackCloud` func in them.

Reply:
I'll make these changes in this PR: https://github.com/Azure/aks-engine/pull/1265

Reply:
this PR is follow up from #1233

Reply:
@jadarsie see this commit:

https://github.com/Azure/aks-engine/pull/1265/commits/76411647e150d09487c412afde88a784846d454f

Basically, we should be gating addons in Azure Stack scenarios during the customData (cloud-init) string concatenation process, not in the "is this addon enabled?" method itself. That latter method we want to keep nice and lean, basically: "do I have this addon configured and enabled in my api model?".

Reply:
Also, as @CecileRobertMichon pointed out, I should have been more opinionated in the prior PR.

Reply:
Sounds good @jackfrancis, just let me know your preferences.

The changes look OK, it looks similar to my first commit: https://github.com/Azure/aks-engine/pull/1233/commits/b8e70377ed7ebe403bd01ab82cab2da0b5691994

Reply:
I think the reason we moved the `!isAzureStack()` condition one level up was to be able to unit test it.